lib/platform/github/types.ts

```diff
@@ -0,0 +1,56 @@
+import GitStorage from '../git/storage';
+import { Pr } from '../common';
+
+// https://developer.github.com/v3/repos/statuses
+// https://developer.github.com/v3/checks/runs/
+export type CombinedBranchState = 'failure' | 'pending' | 'success';
+export type BranchState = 'failure' | 'pending' | 'success' | 'error';
+
+export interface GhBranchStatus {
+  context: string;
+  state: BranchState;
+}
+
+export interface CombinedBranchStatus {
+  state: CombinedBranchState;
+  statuses: GhBranchStatus[];
+}
+
+export interface Comment {
+  id: number;
+  body: string;
+}
+
+export interface GhPr extends Pr {
+  comments: Comment[];
+}
```

Comment:
I think gitea, bitbucket and bitbucket-server have almost exactly the same definition of `Comment`. Move it to the top-level `Pr` definition?

Reply:
sure, this is just a starter to try to fix the `implicitAny` issues first. Then I would like to refactor and optimize more types